amplify/backend/api/colonycdapp/schema.graphql

```diff
@@ -408,7 +409,7 @@ type ColonyAction @model {
   colony: Colony! @belongsTo(fields: ["colonyId"])
   type: ColonyActionType!
   blockNumber: Int!
-  transactionHash: String!
+  isMotion: Boolean
```

Comment:
@willm30 nice, I like that idea, but do you know what's the behaviour of the data field when calling the update mutation? 

The reason I'm asking is because for actions/motions that consist of more than event, you'll need to add some data fields on each of those events. So if the first event creates the action and adds two fields to the data, and then the second event wants to add another few by updating the action, will they be merged or overwritten?

Reply:
Ah, would the problem be that we can't determine whether to merge / overwrite on an action type basis? We'd have to set the merge policy at the level of the model, which might not be appropriate for all action types (i.e. if some need merging and some need overwriting)? 

If we can't do this at a granular, action type level, and it is the case that some will need merging and some overwriting, then I guess we can't do it.

Reply:
I'll share the specific details of the case I'm dealing with right now (and you will most likely do as well):
The Payment action consist of 3 events: `PaymentAdded`, `PayoutClaimed`, `OneTxPaymentMade`
And the data associated with it is:
`
initiator
recipient
amount
tokenAddress
fundamentalChainId
fromDomain
`

`PaymentAdded` will create an action and fill out the `fromDomain` field, using a mutation with input such as:
```gql
  input: {
    id: ...,
    colony: ...,
    data: {
      fromDomain: ...
    }
  }
```

`PayoutClaimed` will update the existing action and provide info on tokenAddress and amount:
```gql
  input: {
    id: ...,
    data: {
      tokenAddress: ...,
      amount: ...
    }
  }
```

My concern is: will this update mutation override the existing `data` field, including the previously defined `fromDomain`? Or will the new fields, `tokenAddress` and `amount` get merged to create an object containing the 3 fields?

I think in any case we'd want it to behave like a standard update function, so if a field is not defined in the input, it should be left intact, and otherwise it should be overwritten.

Reply:
Ok good to know. I'll open an issue for this in actions if that's alright with you, but the approach you outline above looks good to me.